plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java

```diff
@@ -60,6 +63,9 @@
         if (table.isEmpty()) {
             throw new TableNotFoundException(tableName);
         }
+        if (isDeltaLakeTable(table.get())) {
+            throw new TrinoException(HIVE_UNSUPPORTED_FORMAT, "Cannot query Delta Lake table");
+        }
```

Comment:
Same for PartitionsSystemTableProvider

same for Iceberg tables

Reply:
This should throw, just as it does for Delta tables

Reply:
Same here -- you convinced me returning empty is acceptable here, so let's make a prep commit where we do this for existing case (Delta Lake's) first, and Iceberg case being added with consistent handling.

The prep commit message  should capture the domain knowledge we accrued while getting to the bottom of the problem.